rsocket-core/src/main/java/io/rsocket/core/RSocketResponder.java

```diff
@@ -431,55 +398,56 @@ protected void hookOnComplete() {
   }
 
   private void handleStream(int streamId, Flux<Payload> response, int initialRequestN) {
-    response
-        .transform(
-            frameFlux -> {
-              RateLimitableRequestPublisher<Payload> payloads =
-                  RateLimitableRequestPublisher.wrap(frameFlux, Queues.SMALL_BUFFER_SIZE);
-              sendingLimitableSubscriptions.put(streamId, payloads);
-              payloads.request(
-                  initialRequestN >= Integer.MAX_VALUE ? Long.MAX_VALUE : initialRequestN);
-              return payloads;
-            })
-        .subscribe(
-            new BaseSubscriber<Payload>() {
-
-              @Override
-              protected void hookOnNext(Payload payload) {
-                ByteBuf byteBuf;
-                try {
-                  byteBuf = PayloadFrameFlyweight.encodeNext(allocator, streamId, payload);
-                } catch (Throwable t) {
-                  payload.release();
-                  throw Exceptions.propagate(t);
-                }
-
-                payload.release();
-
-                sendProcessor.onNext(byteBuf);
-              }
+    final BaseSubscriber<Payload> subscriber =
+        new BaseSubscriber<Payload>() {
 
-              @Override
-              protected void hookOnComplete() {
-                sendProcessor.onNext(PayloadFrameFlyweight.encodeComplete(allocator, streamId));
-              }
+          @Override
+          protected void hookOnSubscribe(Subscription s) {
+            s.request(initialRequestN >= Integer.MAX_VALUE ? Long.MAX_VALUE : initialRequestN);
+          }
 
-              @Override
-              protected void hookOnError(Throwable throwable) {
-                handleError(streamId, throwable);
-              }
+          @Override
+          protected void hookOnNext(Payload payload) {
+            ByteBuf byteBuf;
+            try {
+              byteBuf = PayloadFrameFlyweight.encodeNext(allocator, streamId, payload);
+            } catch (Throwable t) {
+              payload.release();
+              throw Exceptions.propagate(t);
+            }
 
-              @Override
-              protected void hookFinally(SignalType type) {
-                sendingLimitableSubscriptions.remove(streamId);
-              }
-            });
+            payload.release();
+
+            sendProcessor.onNext(byteBuf);
+          }
+
+          @Override
+          protected void hookOnComplete() {
+            sendProcessor.onNext(PayloadFrameFlyweight.encodeComplete(allocator, streamId));
+          }
+
+          @Override
+          protected void hookOnError(Throwable throwable) {
+            handleError(streamId, throwable);
+          }
+
+          @Override
+          protected void hookFinally(SignalType type) {
+            sendingSubscriptions.remove(streamId);
+          }
+        };
+
+    sendingSubscriptions.put(streamId, subscriber);
+    response.limitRate(Queues.SMALL_BUFFER_SIZE).subscribe(subscriber);
```

Comment:
@rstoyanchev for your attention. I got to rid of the custom rate limiter and use a built-in one